pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -158,21 +158,38 @@ func separateVSHostsAndServices(virtualService model.Config,
 	rule := virtualService.Spec.(*networking.VirtualService)
 	hosts := make([]string, 0)
 	servicesInVirtualService := make([]*model.Service, 0)
+	wchosts := make([]host.Name, 0)
+
+	// As a performance optimization, process non wildcard hosts first, so that they can be
+	// looked up directly in the service registry map.
```

Comment:
Not sure how much this will help, normally we will not have too many hosts

Reply:
There wont be may hosts but for non wildcard hosts we will loop through all the services which is not needed. See the benchmark results and also there was TODO for this also

Reply:
Understood